File: android_native/app/src/main/java/com/example/widgetgallery/presentation/components/AvatarGroup.kt
package com.example.widgetgallery.presentation.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.offset
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

/** Overlapping stacked avatars with a +N overflow. */
@Composable
fun AvatarGroup(
    names: List<String>,
    modifier: Modifier = Modifier,
    max: Int = 4,
    radius: Dp = 20.dp,
) {
    val scheme = MaterialTheme.colorScheme
    val shown = names.take(max)
    val overflow = names.size - shown.size
    val step = radius * 1.4f
    Box(modifier) {
        shown.forEachIndexed { i, name ->
            Box(
                Modifier
                    .offset(x = step * i)
                    .size(radius * 2 + 4.dp)
                    .clip(CircleShape)
                    .background(scheme.surface),
                contentAlignment = Alignment.Center,
            ) {
                Box(
                    Modifier.size(radius * 2).clip(CircleShape)
                        .background(scheme.primaryContainer),
                    contentAlignment = Alignment.Center,
                ) {
                    Text(initialsOf(name), color = scheme.onPrimaryContainer,
                        fontWeight = FontWeight.Bold,
                        fontSize = (radius.value * 0.7f).sp)
                }
            }
        }
        if (overflow > 0) {
            Box(
                Modifier
                    .offset(x = step * shown.size)
                    .size(radius * 2 + 4.dp)
                    .clip(CircleShape)
                    .background(scheme.surface),
                contentAlignment = Alignment.Center,
            ) {
                Box(
                    Modifier.size(radius * 2).clip(CircleShape)
                        .background(scheme.secondaryContainer),
                    contentAlignment = Alignment.Center,
                ) {
                    Text("+$overflow", color = scheme.onSecondaryContainer,
                        fontWeight = FontWeight.Bold,
                        fontSize = (radius.value * 0.7f).sp)
                }
            }
        }
    }
}

File: lib/presentation/widgets/avatar_group.dart
import 'package:flutter/material.dart';

/// Overlapping stacked avatars with a "+N" overflow chip.
class AvatarGroup extends StatelessWidget {
  const AvatarGroup({
    super.key,
    required this.names,
    this.max = 4,
    this.radius = 20,
  });

  final List<String> names;
  final int max;
  final double radius;

  String _initials(String name) {
    final parts = name.trim().split(RegExp(r'\s+'));
    if (parts.isEmpty || parts.first.isEmpty) return '?';
    if (parts.length == 1) return parts.first.characters.first.toUpperCase();
    return (parts.first.characters.first + parts.last.characters.first)
        .toUpperCase();
  }

  @override
  Widget build(BuildContext context) {
    final scheme = Theme.of(context).colorScheme;
    final shown = names.take(max).toList();
    final overflow = names.length - shown.length;
    final step = radius * 1.4;

    return SizedBox(
      height: radius * 2,
      width: step * (shown.length + (overflow > 0 ? 1 : 0)) + radius * 0.6,
      child: Stack(
        children: [
          for (var i = 0; i < shown.length; i++)
            Positioned(
              left: i * step,
              child: CircleAvatar(
                radius: radius + 2,
                backgroundColor: scheme.surface,
                child: CircleAvatar(
                  radius: radius,
                  backgroundColor: scheme.primaryContainer,
                  child: Text(
                    _initials(shown[i]),
                    style: TextStyle(
                      color: scheme.onPrimaryContainer,
                      fontWeight: FontWeight.bold,
                      fontSize: radius * 0.7,
                    ),
                  ),
                ),
              ),
            ),
          if (overflow > 0)
            Positioned(
              left: shown.length * step,
              child: CircleAvatar(
                radius: radius + 2,
                backgroundColor: scheme.surface,
                child: CircleAvatar(
                  radius: radius,
                  backgroundColor: scheme.secondaryContainer,
                  child: Text(
                    '+$overflow',
                    style: TextStyle(
                      color: scheme.onSecondaryContainer,
                      fontWeight: FontWeight.bold,
                      fontSize: radius * 0.7,
                    ),
                  ),
                ),
              ),
            ),
        ],
      ),
    );
  }
}
